src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java

```diff
@@ -416,4 +417,37 @@ default String stringIdRepresentation(@Nullable Object id) {
 		return Objects.toString(id, null);
 	}
 	// endregion
+
+	//region routing
+	/**
+	 * Sets the current {@link RoutingResolver}.
+	 * @param routingResolver must not be {@literal null}
+	 */
+	void setRoutingResolver(RoutingResolver routingResolver);
```

Comment:
I'd suggest to not expose these methods on ther operations interface. They should not be used by the application in the first place.

Reply:
done